### PR TITLE
Add GCE intent based NIC naming tools

### PR DESCRIPTION
**Pull request by humlie**
* Add gce-nic-naming bash script used to determine intent name

* Add 75-gce-network.rules file to call script on specific interface drivers

### CODE REVIEW

**Comment by dorileo** [comment]
I reviewed it internally already, I'm assigning it to someone else to have an extra pair of eyes on it.

**Comment by dorileo** [comment]
@a-crate can you please take a look?

File: src/sbin/gce-nic-naming [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+#   LOG_TAG: Tag to use for logging
+# Arguments:
+#   $@: Info message
+###############################
+function info() {
+  logger -t "${LOG_TAG}" -p local0.info "$@"
+}
+
+###############################
+# Log info message to journal
```
Comment (a-crate) [inline src/sbin/gce-nic-naming:84]: Log notice message to journal?

**Review by a-crate — COMMENTED** [review]
LGTM, would be really great to also incorporate the tests you wrote if it's possible to externalize those.

These files should be added to packaging rules so that it will be included in releases (`packaging/google-compute-engine.spec, `packaging/{debian,ubuntu}/install`). The udev rule will be picked up, but sbin/gce-nic-naming won't.

**Comment by humlie** [comment]
Unfortunately the tests use some internal Google bash scripts so they can't be ported here unless the tests are written with an external test framework.

**Review by a-crate — APPROVED** [review]

**Comment by humlie** [comment]
/hold 
Need to adjust path of script in udev rule.

**Review by a-crate — APPROVED** [review]

**Pull request merged**